activesupport/test/evented_file_update_checker_test.rb

```diff
@@ -77,6 +78,18 @@ def touch(files)
     Process.wait(pid)
   end
 
+  test "can be garbage collected" do
+    previous_threads = Thread.list
+    checker_ref = WeakRef.new(ActiveSupport::EventedFileUpdateChecker.new([], tmpdir => ".rb") { })
+    listener_threads = Thread.list - previous_threads
+
+    wait # Wait for listener thread to start processing events.
+    GC.start
```

Comment:
This test uncovered a race condition in Listen / rb-inotify (see guard/rb-inotify#100) that manifests here as a hang when the [finalizer stops the listener](https://github.com/rails/rails/pull/40133/files#diff-ea498e0b0ed49d4b32a04d0d20c589f6R100).  However, I don't think it's likely that a normal application would trigger the race condition, so I think this PR is safe to merge.  (I've added a `wait` here to prevent the test itself from triggering the race condition.)